packages/instant/src/components/ui/index.ts

```diff
@@ -3,3 +3,5 @@ export { Button } from './button';
 export { Flex } from './flex';
 export { Container } from './container';
 export { Input } from './input';
+export { Icon } from './icon';
+export { Spinner } from './spinner';
```

Comment:
Hrm, I haven't been importing from this `index` file, just importing directly from the components themselves.  What is the accepted pattern here?

Reply:
I can see how it seems arbitrary... I don't have great reasons for why I went with this pattern. I guess I saw it as kind of a package (like a styleguide) within this package, so structured it that way. Also it's convenient because they are our most used components, and are often used together. Think we should be consistent with this so fixed the other imports. Let me know what you think.

Reply:
I could understand this being convenient if the ui components were exposed as a npm package that someone else would be importing from.

But, upon thinking about this more, I worry that this approach could be bad for code splitting and tree shaking.  Let's say we only use the Spinner and no other ui components in our entry point.  I worry that the code splitting and tree shaking tooling would think that there could be side effects in `ui/index.ts`, and would include all the other ui components in the build.

For that reason, I'd refer that we import directly from the individual component files instead of using this `index` pattern.  It also reduces some boilerplate

Reply:
Ok before going deeper down this "if it was an npm package" hypothetical, I'd just like to say it's not and probably won't be :P. As mentioned I don't have great reasons except that I was thinking about it that way and that it is a bit nicer to import them this way as they are our building blocks for the other things in `components/`. It really is mostly a stylistic / convenience thing. 

In terms of the tree shaking discussion, I think it will work as long as your use the `export` `import` syntax which we are in this case (but not in the case where we are exporting a single object, which I agree is not great for tree shaking although I am a fan of the syntax / style). Also I don't think webpack tries to figure out if there are side-effects, you need to tell it whether there are or not.

Reply:
Def agree RE: npm package, and that is a good point about the webpack `sideEffects` config!

Overall I relate to the thought that you've brought up in other scenarios, in which there should be one accepted way to do things.  I'd prefer not to have to think about: "Do I import this component directly from the file, or do I import it from its' parent `index`? " 

I personally don't feel like this index pattern makes things really any more convenient, and it introduces inconsistencies with how we import objects, and adds additional (although small) overhead to introducing a new UI component.

For the sake of keeping things simple and consistent, I feel pretty strongly that we should just import the components directly.

Reply:
Chatted in Slack. While I do prefer the `./iui` pattern, @steveklebanoff feels pretty strongly it should be a direct import, brings up some good points, and is willing to make the change, so we will follow up on this in a follow-up branch :)